tracing-appender/src/rolling/compression.rs

```diff
@@ -0,0 +1,172 @@
+    GzipLevel6,
+    /// Gzip compression level 7
+    GzipLevel7,
+    /// Gzip compression level 8
+    GzipLevel8,
+    /// Gzip compression level 9
+    GzipLevel9,
+}
+
+impl Into<CompressionConfig> for CompressionOption {
+    fn into(self) -> CompressionConfig {
+        match self {
+            CompressionOption::GzipNone => CompressionConfig::Gzip(GzipCompression {
+                level: GzipCompressionLevel::Literal(GzipCompressionLevelLiteral::None),
+            }),
+            CompressionOption::GzipFast => CompressionConfig::Gzip(GzipCompression {
+                level: GzipCompressionLevel::Literal(GzipCompressionLevelLiteral::Fast),
+            }),
+            CompressionOption::GzipBest => CompressionConfig::Gzip(GzipCompression {
+                level: GzipCompressionLevel::Literal(GzipCompressionLevelLiteral::Best),
+            }),
+            CompressionOption::GzipLevel0 => CompressionConfig::Gzip(GzipCompression {
+                level: GzipCompressionLevel::Numerical(GzipCompressionLevelNumerical::Level0),
+            }),
+            CompressionOption::GzipLevel1 => CompressionConfig::Gzip(GzipCompression {
+                level: GzipCompressionLevel::Numerical(GzipCompressionLevelNumerical::Level1),
+            }),
+            CompressionOption::GzipLevel2 => CompressionConfig::Gzip(GzipCompression {
+                level: GzipCompressionLevel::Numerical(GzipCompressionLevelNumerical::Level2),
+            }),
+            CompressionOption::GzipLevel3 => CompressionConfig::Gzip(GzipCompression {
+                level: GzipCompressionLevel::Numerical(GzipCompressionLevelNumerical::Level3),
+            }),
+            CompressionOption::GzipLevel4 => CompressionConfig::Gzip(GzipCompression {
+                level: GzipCompressionLevel::Numerical(GzipCompressionLevelNumerical::Level4),
+            }),
+            CompressionOption::GzipLevel5 => CompressionConfig::Gzip(GzipCompression {
+                level: GzipCompressionLevel::Numerical(GzipCompressionLevelNumerical::Level5),
+            }),
+            CompressionOption::GzipLevel6 => CompressionConfig::Gzip(GzipCompression {
+                level: GzipCompressionLevel::Numerical(GzipCompressionLevelNumerical::Level6),
+            }),
+            CompressionOption::GzipLevel7 => CompressionConfig::Gzip(GzipCompression {
+                level: GzipCompressionLevel::Numerical(GzipCompressionLevelNumerical::Level7),
+            }),
+            CompressionOption::GzipLevel8 => CompressionConfig::Gzip(GzipCompression {
+                level: GzipCompressionLevel::Numerical(GzipCompressionLevelNumerical::Level8),
+            }),
+            CompressionOption::GzipLevel9 => CompressionConfig::Gzip(GzipCompression {
+                level: GzipCompressionLevel::Numerical(GzipCompressionLevelNumerical::Level9),
+            }),
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+
+    #[test]
+    fn test_compression() {}
+}
```

Comment:
if we aren't actually going to write any tests for this, i think this should probably just be removed?
```suggestion
```

Reply:
Definitely test is needed :)